community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -864,15 +977,25 @@ public PrimitiveIntIterator nodeGetPropertyKeys( KernelStatement state, long nod
                 throw new EntityNotFoundException( EntityType.NODE, nodeId );
             }
 
-            return Cursors.intIterator( nodeCursor.properties(), PropertyCursor.GET_KEY_INDEX_ID );
+            PrimitiveIntStack keys = new PrimitiveIntStack( 16 );
+            try (PropertyCursor properties = nodeCursor.properties())
```

Comment:
Can we extract following
``` java 
            PrimitiveIntStack keys = new PrimitiveIntStack( 16 );
            try (PropertyCursor properties = cursor.properties())
            {
                while (properties.next())
                {
                    keys.push( properties.propertyKeyId() );
                }
            }

            return keys.iterator();
```
to have helper method that will return cursor properties key iterator and reuse it for example in https://github.com/rickardoberg/neo4j/blob/propertycursor/community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java#L1046